packages/engine/src/datastore/schema/field_spec/mod.rs

```diff
@@ -52,21 +56,21 @@ pub fn create_field_key(scope: FieldScope, name: &str, source: FieldSource) -> R
 }
 
 /// Defines the source from which a Field was specified, useful for resolving clashes
+// TODO: Find a good name, which does not conflict with `FieldSource`
 #[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
-pub enum FieldSource {
+pub enum Source {
```

Comment:
We could call it an EngineComponent or something?

Can you have another think along those lines, maybe you can come up with something better

Reply:
I think `EngineComponent` is a decent name for `Source`. Currently, I don't know a better name, but I will think about it a bit more.